xla/service/cpu/ir_emitter.cc

```diff
@@ -3373,7 +3373,17 @@ void IrEmitter::TracingState::EmitTracingStart(llvm::IRBuilder<>* b,
     fn->setDoesNotThrow();
     fn->setOnlyAccessesArgMemory();
   }
-  auto* hlo_name = b->CreateGlobalStringPtr(hlo->name());
+
+  // Pass opcode as argument to TraceMe call
+  auto hlo_name_str = hlo->name();
+  if (hlo->opcode() == HloOpcode::kCustomCall) {
+    // For custom call, passing custom call target is more informative
+    hlo_name_str = hlo->custom_call_target();
```

Comment:
For custom call, it's probably still good to know the custom call id. Having `custom-call.<id>` as prefix should still be helpful, e.g., `custom-call.46.__onednn$matmul`.